tmd/areas/create_area_weights.py

```diff
@@ -151,15 +145,38 @@ def loss_function_value(wght, target_matrix, target_array):
     return 0.5 * np.sum(np.square(act_minus_exp))
 
 
+def objective_function(x, *args):
+    """
+    Objective function for minimization of sum of squared deviations.
+    """
+    A, b, lambda_ = args
+    target_deviations = A @ x - b  # JAX sparse matrix-vector multiplication
+    weight_deviations = jnp.sqrt(lambda_) * (x - 1)
+    deviations = jnp.concatenate([target_deviations, weight_deviations])
+    return jnp.sum(jnp.square(deviations))
```

Comment:
Here's suggested expanded documentation for the objective function, for your consideration:


def objective_function(x, *args):
    """
    Objective function for the minimization problem:
    
    Minimize the sum of squared deviations between Ax and b, 
    plus a regularization term that penalizes deviations of x from 1.
    
    The objective function has two main components:
    1. Target deviations: (Ax - b), which measures how well the vector x satisfies the linear constraints represented by A and b.
    2. Regularization term: sqrt(lambda) * (x - 1), which penalizes deviations of the vector x from 1.
    
    Args:
        x (array): The optimization variable.
        *args: Additional arguments, expected to be (A, b, lambda_):
            A (BCOO): A sparse matrix representing the linear system.
            b (array): A target vector.
            lambda_ (float): Regularization parameter.
            
    Returns:
        float: The sum of squared deviations from the target and regularization.
    """

Reply:
Done, but less elaborately.